src/OpenApi/gen/XmlCommentGenerator.Emitter.cs

```diff
@@ -80,148 +80,198 @@ file record XmlComment(
+        /// <summary>
+        /// Generates a documentation ID string for a type.
+        /// This method handles nested types (replacing '+' with '.'),
+        /// generic types, arrays, pointers, by-ref types, and generic parameters.
+        /// The <paramref name="includeGenericArguments"/> flag controls whether
+        /// constructed generic type arguments are emitted, while <paramref name="omitGenericArity"/>
+        /// controls whether the generic arity marker (e.g. "`1") is appended.
+        /// </summary>
+        private static string GetTypeDocId(Type type, bool includeGenericArguments, bool omitGenericArity)
         {
-            if (methodInfo is null)
+            if (type.IsGenericParameter)
             {
-                return Cache.TryGetValue(new MemberKey(type, MemberType.Property, null, null, null), out xmlComment);
+                // Use `` for method-level generic parameters and ` for type-level.
+                if (type.DeclaringMethod != null)
+                {
+                    return "``" + type.GenericParameterPosition;
+                }
+                else if (type.DeclaringType != null)
+                {
+                    return "`" + type.GenericParameterPosition;
+                }
+                else
+                {
+                    return type.Name;
+                }
             }
 
-            return Cache.TryGetValue(MemberKey.FromMethodInfo(methodInfo), out xmlComment);
-        }
+            if (type.IsArray && type.GetElementType() is { } elementType)
+            {
+                int rank = type.GetArrayRank();
+                string elementTypeId = GetTypeDocId(elementType, includeGenericArguments, omitGenericArity);
+                return rank == 1 ? elementTypeId + "[]" : elementTypeId + "[" + new string(',', rank - 1) + "]";
+            }
 
-        internal static bool TryGetXmlComment(Type? type, string? memberName, [NotNullWhen(true)] out XmlComment? xmlComment)
-        {
-            return Cache.TryGetValue(new MemberKey(type, memberName is null ? MemberType.Type : MemberType.Property, memberName, null, null), out xmlComment);
+            if (type.IsGenericType)
+            {
+                // Get the generic type definition.
+                Type genericDef = type.GetGenericTypeDefinition();
+                string defName = genericDef.FullName != null
+                    ? genericDef.FullName.Replace('+', '.')
+                    : genericDef.Name;
+                // Remove any existing generic arity marker.
+                int backtickIndex = defName.IndexOf('`');
+                if (backtickIndex >= 0)
+                {
+                    defName = defName.Substring(0, backtickIndex);
+                }
+                // Append the generic arity unless omitted.
+                if (!omitGenericArity)
+                {
+                    int arity = genericDef.GetGenericArguments().Length;
+                    defName += "`" + arity;
```

Comment:
I would err on the side of caution here and format the arity explicitly with `CultureInfo.InvariantCulture` so that users on systems configured for non-English won't potentially get incorrect number formatting (e.g. scenario of `1000` becoming `1,000` or `1.000`). Might be some other occurrences of this elsewhere too.